src/envoy/utils/authn.cc

```diff
@@ -58,12 +58,14 @@ void Authentication::SaveAuthAttributesToStruct(
                   origin.audiences(0));
     }
     if (!origin.groups().empty()) {
```

Comment:
Can we do it for all claims instead of just "groups"?

Reply:
A JWT may contain various claims, including user defined ones (e.g., department, machine-model, IMEI and etc). Probably we can place all claims in a JWT into request.auth.claims[] or we can all non-well-known claims in a JWT into request.auth.claims[]. But such a PR can be a separate PR to keep this PR small and simple.

Reply:
In the Pilot PR https://github.com/istio/istio/pull/7747 you just checked in, all claims are matched using ListMatcher. If you don't create ListValue for all claims here, RBAC will break.

Reply:
Per discussions, in this PR, JWT claims are extracted as string lists for RBAC list matcher. A new PR https://github.com/istio/proxy/issues/1920 has been created to change the "claims" to a protobuf struct.